app/main/views/suppliers.py

```diff
@@ -195,6 +152,194 @@ def update_supplier_name(supplier_id):
     return redirect(url_for('.supplier_details', supplier_id=supplier_id))
 
 
+@main.route('/suppliers/<int:supplier_id>/edit/registered-name', methods=['GET', 'POST'])
+@role_required('admin-ccs-data-controller')
+def edit_supplier_registered_name(supplier_id):
+    supplier = data_api_client.get_supplier(supplier_id)['suppliers']
+
+    # Get SupplierFrameworks for frameworks the role is interested in, sorted by oldest frameworkLiveAtUTC first
+    supplier_framework_interests = data_api_client.get_supplier_frameworks(supplier_id)["frameworkInterest"]
+    supplier_frameworks = interesting_frameworks(supplier_framework_interests, current_user, data_api_client)
```

Comment:
We see this pattern a lot, where we get the framework interests and then filter using `interesting_frameworks`; is it worth adding the initial step into the helper?

Reply:
Potentially? I'm not super keen on the general pattern of passing the instantiated `data_api_client` through to a helper if we're using it at the view level as well, as it can often make test patching confusing.

Reply:
Hmm, the helper already uses the data_api_client?

Reply:
🤦‍♀️ It probably shouldn't! And that's why this is a WIP...